agent/stats/queue.go

```diff
@@ -64,7 +65,16 @@ func (queue *Queue) Add(rawStat *ContainerStats) {
 	if queueLength != 0 {
 		// % utilization can be calculated only when queue is non-empty.
 		lastStat := queue.buffer[queueLength-1]
-		stat.CPUUsagePerc = 100 * float32(rawStat.cpuUsage-lastStat.cpuUsage) / float32(rawStat.timestamp.Sub(lastStat.Timestamp).Nanoseconds())
+		timeSinceLastStat := float32(rawStat.timestamp.Sub(lastStat.Timestamp).Nanoseconds())
+		if timeSinceLastStat == 0 {
```

Comment:
Minor preference to invert this: positive case in the `if` and negative case in the `else`.